airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -232,12 +232,11 @@ def __init__(self,
         self.customer_managed_key = customer_managed_key
         self.single_node = num_workers == 0
 
-        assert not (self.custom_image and self.image_version), \
-            "custom_image and image_version can't be both set"
+        if self.custom_image and self.image_version:
+            raise ValueError("The custom_image and image_version can't be both set")
 
-        assert (
-            not self.single_node or (self.single_node and self.num_preemptible_workers == 0)
-        ), "num_workers == 0 means single node mode - no preemptibles allowed"
+        if self.num_preemptible_workers == 0 and not self.single_node:
+            raise ValueError("num_workers == 0 means single node mode - no preemptibles allowed")
```

Comment:
This doesn't cover the case when `self.num_preemptible_workers >= 0 and self.single_node`

So I think it should be
```suggestion
        if (self.single_node and self.num_preemptible_workers > 0):
            raise ValueError("num_workers == 0 means single node mode - no preemptibles allowed")
```

Reply:
I was quite a bit confused with the condition :). But you are right:

When you convert assert to if you need to revert the original condition. If you do it in the above case using DeMorgan you get:
`not  (not self.single_node or (self.single_node and self.num_preemptible_workers == 0))`

which simplifies to:

`self.single_node and (not self.single_node or self.num_preemptible_workers != 0)`

It will only work indeed when self.single_node is true and self.num_preemptible_workers >0

Reply:
I will change it and also update the error message because it is really confusing :)